docs/adrs/adr-0001-dependency_management.md

```diff
@@ -0,0 +1,53 @@
+
+* : Dependabot already used in Jetstream
+* +: Jetstream uses it already so there is an existing base of experties in the data team
+* +: Dependencies can be automatically updated with a script like the [update_deps](https://github.com/mozilla/jetstream/blob/main/script/update_deps) script in Jetstream
+* -: Still using pip's dependency resolver, which is fast but can have issues with incompatible versions.  This may be mitigated or even solved by the way pip-tools pins dependencies.
+* -: Jetstream doesn't use `pyproject.toml` for configuring dependencies, though it does seem to be supported by `pip-tools`
+
+### 2. poetry
+(poetry)[https://python-poetry.org/] is a popular tool for dependency managment.  When starting development with a poetry project, one uses the `poetry install` command to create a virtual environment for the project and install the package and its dependencies.  New dependencies are added with the `poetry add` command.  poetry (and other tools) are configured in the `pyproject.toml`.
+
+* : Can add Dependabot, which is alrady built-in to github tool that has additional uses like checking for security vulnerabilities
+* +: Very reliable dependency resolution
+* +: Uses `pyproject.toml`, which is the new standard and is nice for consolidating other config files
+* +: Also acts as a virtualenv manager
+* +: No changes to user workflow
+* +: When a new dependency is added with `poetry add` the `poetry.lock` and `pyproject.toml` files are automatically updated so there minimal risk of a new dependency not being accounted for
+* -: Dependency resolution is slow, when you add a new package it can take several minutes to figure it out
+* -: Developers would have to modify their workflow to accomodate poetry's virtualenv management
+* -: Not wildely use in the data team
+* -: would need to update tox to use poetry to build and publish package
+
+### 3. mamba/conda
+(mamba)[https://mamba.readthedocs.io/en/latest/] is a fast, robust, and cross-platform package manager. It is fully compatible with conda, which is a common package manager in Python, and can be a drop-in replacement for conda. Mamba creates a virtual environment that is isolated from each other.
+
+* Creates isolated virtual environment that encapsulates all the packages in its own designated space
+* Can assign a specific Python version to the given virtual environment
+* Dependency resoluion is fast and optimized for parallelism
+* Uses an `environment.yml` to specify dependencies
+* Can consume `pyproject.toml` when specifying packages
```

Comment:
A `-` we should note is that we'd have to come up with our own script to do the automatic update based on the venv like Jetstream does

Reply:
A couple other negatives:
- we note that poetry is not widely used in the data team, but is conda used in any mozilla data repos? I only regularly work with around a dozen repos and those use pip-tools and poetry but not conda/mamba
- I believe the storage size of a mozanalysis environment would drastically increase, at least from what I remember of working with conda. Doesn't it come with a lot of libraries pre-packaged?
- the isolated virtual environment comes with potential for confusion over system libraries vs conda libraries as developers move between projects

I'm intentionally searching for negatives here to ensure we're considering everything (I think @teonbrooks 
covered the positives pretty well already). I have used conda in the past and found a lot to like about it, but also some frustrations. That said, I wasn't aware of poetry at the time -- what are the advantages vs poetry for our situation?

Reply:
@mikewilli to answer your question, conda environments do not come with any packages pre-installed, it comes with the default or specified python version and that's pretty much it. 

with Conda there is some confusion with the Anaconda and the bundled solution they offer. They do use conda as a package manager but it also comes with a host of pre-configured packages.

Reply:
Thanks for the update Teon.  After looking it over I still think pip-tools is the best option just because of familiarity.  In the end I think it's pretty similar to the poetry vs pip-tools decision.  It looks like the big advantage of conda is a speedier dependency resolver and the built-in virtualenvs, and I don't think that outweighs the cost of having to figure out how to rework the Jetstream scripts to use it.